components/script/dom/userscripts.rs

```diff
@@ -40,13 +40,11 @@ pub fn load_script(head: &HTMLHeadElement) {
         files.sort();
 
         for file in files {
-            let name = match file.into_os_string().into_string() {
-                Ok(ref s) if s.ends_with(".js") => "file://".to_owned() + &s[..],
-                _ => continue
-            };
-            let new_script = doc.CreateElement(DOMString::from("script")).unwrap();
-            new_script.set_string_attribute(&local_name!("src"), DOMString::from(name));
-            node.InsertBefore(new_script.upcast(), first_child.r()).unwrap();
+            let mut f = File::open(&file).unwrap();
+            let mut contents = vec![];
+            f.read_to_end(&mut contents).unwrap();
+            let script_text = String::from_utf8(contents).unwrap();
+            win.upcast::<GlobalScope>().evaluate_js_on_global_with_result(&script_text, rval.handle_mut());
```

Comment:
@nox, does this line look right? (It doesn’t look particularly wrong to me, I just don’t know.)